src/autocomplete/shared/compute-token-position.ts

```diff
@@ -0,0 +1,116 @@
+import {ParseTree, ParserRuleContext, TerminalNode, Token, TokenStream} from 'antlr4ng';
+
+import {CursorPosition} from './autocomplete-types';
+
+type TokenPosition = {index: number; context: ParseTree; text: string};
```

Comment:
this interface doesn't look like `TokenPosition` to me
I think position is only a small part of it, maybe we can rename it to something more appropriate?
what about `TokenContext`? or even `CursorContext`?